components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/publish/choreo/ChoreoStatisticsPublisher.java

```diff
@@ -0,0 +1,293 @@
+                    return;
+                }
+                if ((StatisticsConstants.FLOW_STATISTICS_API.equals(event.getComponentType()) &&
+                                analyticsEnabledForAPI) ||
+                        (StatisticsConstants.FLOW_STATISTICS_SEQUENCE.equals(event.getComponentType()) &&
+                                analyticsEnabledForSequences) ||
+                        (StatisticsConstants.FLOW_STATISTICS_ENDPOINT.equals(event.getComponentType()) &&
+                                analyticsEnabledForEndpoints) ||
+                        (StatisticsConstants.FLOW_STATISTICS_INBOUNDENDPOINT.equals(event.getComponentType()) &&
+                                analyticsEnabledForInboundEndpoints) ||
+                        (StatisticsConstants.FLOW_STATISTICS_PROXYSERVICE.equals(event.getComponentType()) &&
+                                analyticsEnabledForProxyServices)) {
+                    if ((StatisticsConstants.FLOW_STATISTICS_SEQUENCE.equals(event.getComponentType()) &&
+                            event.getElasticMetadata().getSequence(event.getComponentName()) == null) ||
+                            (StatisticsConstants.FLOW_STATISTICS_ENDPOINT.equals(event.getComponentType()) &&
+                            event.getElasticMetadata().getEndpoint(event.getComponentName()) == null)) {
+                        // If Endpoint or Sequence is null
+                        return;
+                    }
+                    publishAnalytics(event);
+                }
+            });
+        }
+    }
+
+    /**
+     * Publish the analytics.
+     *
+     * Publish the analytics to Event Hub (For Choreo Analytics).
+     *
+     * @param event This is the event published by synapse.
+     */
+    private void publishAnalytics(PublishingEvent event) {
+        JsonObject payload = choreoPayload(event);
+        // In here, publishing data to event hub should be handled.
+        // For example,
+        // builder.addAttribute(entry.getKey(), entry.getValue());
+        // counterMetric.incrementCount(builder);
+        // Where entry has assigned by payload data.
+        // https://github.com/wso2/apim-analytics-publisher handles API-M analytics publishing for Choreo.
+        // Should be able to reuse the same method to publish data to event hub with some changes.
+        log.info("Choreo Analytics is enabled. Payload: " + payload);
+    }
+
+    /**
+     * Generate the payload.
+     *
+     * Generate the payload to send to Event Hub (For Choreo Analytics).
+     *
+     * @param event This is the event published by synapse.
+     * @return This returns a JsonObject which contains the required flat Json by Choreo.
+     */
+    private JsonObject choreoPayload (PublishingEvent event) {
+        try {
+            ElasticMetadata metadata = event.getElasticMetadata();
+            JsonObject pL = new JsonObject();
```

Comment:
Let's rename `pL` to `payload`

Reply:
Renamed by https://github.com/wso2/micro-integrator/pull/2927/commits/7ce08a2dbe17617040a9584d9afe40c25a4178fc